src/registrar/forms/domain_request_wizard.py

```diff
@@ -22,10 +22,151 @@
 
 
 class RequestingEntityForm(RegistrarForm):
-    organization_name = forms.CharField(
-        label="Organization name",
+    """The requesting entity form contains a dropdown for suborganizations,
+    and some (hidden by default) input fields that allow the user to request for a suborganization.
+    All of these fields are not required by default, but as we use javascript to conditionally show
+    and hide some of these, they then become required in certain circumstances."""
+
+    # IMPORTANT: This is tied to DomainRequest.is_requesting_new_suborganization().
+    # This is due to the from_database method on DomainRequestWizard.
+    # Add a hidden field to store if the user is requesting a new suborganization.
+    # This hidden boolean is used for our javascript to communicate to us and to it.
+    # If true, the suborganization form will auto select a js value "Other".
+    # If this selection is made on the form (tracked by js), then it will toggle the form value of this.
+    # In other words, this essentially tracks if the suborganization field == "Other".
+    # "Other" is just an imaginary value that is otherwise invalid.
+    # Note the logic in `def clean` and line 2744 in get-gov.js
```

Comment:
**comment**: don't include line numbers in code comments please, they will fall out of date